transactions/README.md

```diff
@@ -142,17 +141,96 @@ This schedule is conflict-serializable because there is an edge from transfer to
 
 ![serial schedule image](/transactions/cyclic.png)  
 
-## 2PL Permits Interleaved Access
+## 2 PL Only Allows Serializable Schedules
+The below example is precluded because Sum cannot grab the S-lock on B after it has released the S-lock on A.
+
+![serial schedule image](/transactions/2pl_example.png)  
 
-TODO: Summarize (preferably with ASCII art) L15 slides 37
+## 2PL Permits Interleaved Access
+Key rule is that once a transaction has released a lock, it is not allowed to obtain any other locks. Progression of time goes downward in the chart.
+
+```
+Transfer Transaction			Sum Transaction
+								ACQ S-Lock on A
+								Read A = 1
+ACQ S-Lock on A
+Read A = 1
+								ACQ S-Lock on B
+								Read B = 1
+								Release Locks
+								Commit (2)
+ACQ X-Lock on A
+Write A = 2
+ACQ S-Lock on B
+Read B = 1
+ACQ X-Lock on B
+Write B = 2
+Release Locks
+Commit
+```
+
+The following schedule could also have occurred, but in this case Sum appears to have occurred AFTER the transaction.
+
+```
+Transfer Transaction			Sum Transaction
+ACQ S-Lock on A
+Read A = 1					
+ACQ X-Lock on A
+Write A = 2
+ACQ S-Lock on B
+Read B = 1
+ACQ X-Lock on B
+Write B = 2
+Release Locks																	ACQ S-Lock on A
+								Read A = 2
+								ACQ S-Lock on B
+								Read B = 2
+								Release Locks
+								Commit (4)
+Commit
+```
 
 ## 2PL Doesn't Exploit Full Concurrency
-
-TODO: Summarize (preferably with ASCII art) L15 slides 38
+However, as noted before, 2PL precludes some serializable schedules. Events in parentheses are not allowed under 2PL, but must happen in order for this schedule to occur.
+
+```
+Transfer Transaction			Sum Transaction
+ACQ S-Lock on A
+Read A = 1
+ACQ X-Lock on A						
+Write A = 2
+								(ACQ S-Lock on A)
+								Read A = 2
+ACQ S-Lock on B
+Read B = 1
+ACQ X-Lock on B
+Write B = 2
+Release Locks	
+Commit
+								(ACQ S-Lock on A)
+								Read B = 2
+								Commit (4)
+```
 
 ## Undisciplined Locking causes Non-serializable Schedules
 
-TODO: Give example (preferably with ASCII art) of the above, and how it would be detected.
+(1) T1 locks all pages containing volleyball players that
+play setter, and finds tallest setter (height = 6').
+	
+(2) T2 inserts a new player that is a setter, (height = 6' 1")
```

Comment:
How can T2 insert the new setter if all setter-pages are locked? Maybe there's one player per page, in which case it probably makes more sense to call it a record.